gremlin-go/driver/graphBinary.go

```diff
@@ -646,6 +658,53 @@ const (
 	valueFlagNone byte = 0
 )
 
+func enumWriter(value interface{}, buffer *bytes.Buffer, typeSerializer *graphBinaryTypeSerializer) ([]byte, error) {
+	_, err := typeSerializer.write(reflect.ValueOf(value).String(), buffer)
+	return buffer.Bytes(), err
+}
+
+func pWriter(value interface{}, buffer *bytes.Buffer, typeSerializer *graphBinaryTypeSerializer) ([]byte, error) {
+	v := value.(*p)
+	_, err := typeSerializer.writeValue(v.operator, buffer, false)
+	if err != nil {
+		return nil, err
+	}
+
+	err = binary.Write(buffer, binary.BigEndian, int32(len(v.values)))
+	if err != nil {
+		return nil, err
+	}
+
+	for _, pValue := range v.values {
+		_, err := typeSerializer.write(pValue, buffer)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return buffer.Bytes(), err
+}
+
+func textPWriter(value interface{}, buffer *bytes.Buffer, typeSerializer *graphBinaryTypeSerializer) ([]byte, error) {
+	v := value.(*textP)
+	_, err := typeSerializer.writeValue(v.operator, buffer, false)
+	if err != nil {
+		return nil, err
+	}
+
+	err = binary.Write(buffer, binary.BigEndian, int32(len(v.values)))
+	if err != nil {
+		return nil, err
+	}
+
+	for pValue := range v.values {
+		_, err := typeSerializer.write(pValue, buffer)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return buffer.Bytes(), err
+}
```

Comment:
Do these need readers as well?

Reply:
No we do not accept predicates, we only write them.